### PR TITLE
Rename sug parsers and commands

### LINKED ISSUE #471: Rename Sug Parsers and Commands

### PR DESCRIPTION
**Pull request by johannagwan**
Closes #471

### CODE REVIEW

File: src/main/java/com/notably/logic/suggestion/SuggestionEngine.java [hunk 1]
```diff
@@ -99,34 +100,35 @@ private void suggest(String userInput) {
...
+            return new HelpSuggestionHandler(model).handle();
 
-        case ExitSuggestionCommandParser.COMMAND_WORD:
-            return new ExitSuggestionCommandParser(model).parse(arguments);
+        case ExitSuggestionHandler.COMMAND_WORD:
+            return new ExitSuggestionHandler(model).handle();
 
         default:
-            return new ErrorSuggestionCommandParser(model).parse(arguments);
+            return new ErrorSuggestionArgHandler(model).handleArg(arguments);
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/suggestion/SuggestionEngine.java]: Actually, this default path will never get executed right? I'm not sure if that's the case. But if it is, would it make sense to not have an `ErrorSuggestionArgHandler` and throw an `AssertionError` here instead?

File: src/main/java/com/notably/logic/suggestion/handler/SearchSuggestionArgHandler.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ public SearchSuggestionCommandParser(Model model) {
             return Optional.empty();
         } else {
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/suggestion/handler/SearchSuggestionArgHandler.java]: FYI, you don't really need the `else` branch here!

File: src/main/java/com/notably/logic/suggestion/handler/SuggestionArgHandler.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Represents a Suggestion Argument Handler that is able to handle user input into a SuggestionGenerator.
+ *
+ * @param <T> Generic type of SuggestionGenerator.
+ */
+public interface SuggestionArgHandler<T extends SuggestionGenerator> {
+    /**
+     * Handles user input into a suggestion generator and returns it.
+     *
+     * @param userInput The user input.
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/suggestion/handler/SuggestionArgHandler.java:16]: Is it more descriptive to say `The arguments part of the user input`, then explains a bit on what you mean by the `arguments` part in the interface description?

File: src/main/java/com/notably/model/suggestion/SuggestionItem.java [hunk 4]
```diff
@@ -14,9 +14,9 @@
     Optional<String> getProperty(String key);
 
     /**
-     * Executes the action taken after the suggestion item is chosen.
-     * The action will be different depending on the type of SuggestionCommand.
-     * @return The action taken after the suggestion item is chosen.
+     * Executes the action taken after the generator item is chosen.
```
Comment (kevinputera) [inline src/main/java/com/notably/model/suggestion/SuggestionItem.java]: Hmmm I feel that the original JavaDoc is more explanatory though

File: src/test/java/com/notably/logic/parser/ArgumentTokenizerTest.java [hunk 5]
```diff
@@ -7,6 +7,10 @@
 
 import org.junit.jupiter.api.Test;
 
+import com.notably.commons.parser.ArgumentMultimap;
+import com.notably.commons.parser.ArgumentTokenizer;
+import com.notably.commons.parser.Prefix;
+
 
 public class ArgumentTokenizerTest {
```
Comment (kevinputera) [inline src/test/java/com/notably/logic/parser/ArgumentTokenizerTest.java:10]: This class should be under `com.notably.commons.parser`

**Review by kevinputera — APPROVED** [review]
Overall, LGTM! Some comments below

File: src/main/java/com/notably/logic/suggestion/SuggestionEngine.java (hunk 1 above)
Comment (johannagwan) [reply src/main/java/com/notably/logic/suggestion/SuggestionEngine.java]: ok!

File: src/main/java/com/notably/logic/suggestion/handler/SuggestionArgHandler.java (hunk 3 above)
Comment (johannagwan) [reply src/main/java/com/notably/logic/suggestion/handler/SuggestionArgHandler.java:16]: Ok!

File: src/test/java/com/notably/logic/parser/ArgumentTokenizerTest.java (hunk 5 above)
Comment (johannagwan) [reply src/test/java/com/notably/logic/parser/ArgumentTokenizerTest.java:10]: Thanks for spotting it!

**Pull request merged**